pydantic_ai_slim/pydantic_ai/_agent_graph.py

```diff
@@ -649,7 +649,11 @@ async def process_function_tools(  # noqa C901
                 )
                 output_parts.append(part)
         else:
-            output_parts.append(_unknown_tool(call.tool_name, call.tool_call_id, ctx))
+            yield _messages.FunctionToolCallEvent(call)
+
+            result = _unknown_tool(call.tool_name, call.tool_call_id, ctx)
+            yield _messages.FunctionToolResultEvent(result, tool_call_id=call.tool_call_id)
```

Comment:
I think we'll want to do the same thing for the `ToolReturnPart` we output on line 645